go/cmd/dolt/commands/sql.go

```diff
@@ -299,19 +314,39 @@ func execQuery(sqlCtx *sql.Context, dEnv *env.DoltEnv, root *doltdb.RootValue, q
 
 	if rowIter != nil {
 		defer rowIter.Close()
-		err = se.prettyPrintResults(sqlCtx, se.ddb.ValueReadWriter().Format(), sqlSch, rowIter)
+		err = se.prettyPrintResults(sqlCtx, sqlSch, rowIter)
 		if err != nil {
 			return nil, errhand.VerboseErrorFromError(err)
 		}
 	}
 
-	newRoot, err := se.sdb.GetRoot(sqlCtx)
-
+	newRoots, err := se.getRoots(sqlCtx)
 	if err != nil {
-		return nil, errhand.BuildDError("Failed to get root").AddCause(err).Build()
+		return nil, errhand.BuildDError("failed to get roots").AddCause(err).Build()
 	}
 
-	return newRoot, nil
+	return newRoots, nil
+}
+
+// CollectDBs takes a MultiRepoEnv and creates Database objects from each environment and returns a slice of these
+// objects.
+func CollectDBs(mrEnv env.MultiRepoEnv, roots map[string]*doltdb.RootValue, batchMode bool) []dsqle.Database {
```

Comment:
Would prefer dbFactory fn type alias instead of a bool. Then you can just pass the appropriate method ref at call sites.

Reply:
done